config/software/opensearch.rb

```diff
@@ -0,0 +1,42 @@
+#
+# Copyright 2020 Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name "opensearch"
+default_version "1.2.4"
+
+dependency "server-open-jre"
+
+license "Apache-2.0"
+license_file "LICENSE.txt"
+skip_transitive_dependency_licensing true
+
+source url: "https://opensearch.org/source.html"
```

Comment:
I think, We should give the full source URL to download.Like this - https://github.com/chef/chef-server/blob/dd471001196a0d41b8471648f6941a285fa6eb94/omnibus/config/software/opensearch.rb
Have you verified it by your testing whether omnibus is able to build opensearch?

Reply:
I dont think complete source-URL is required, reason being is, I can see that it is successfully getting downloaded.
As seen in the below link
https://buildkite.com/chef-oss/chef-omnibus-software-main-verify/builds/1675#ce72a659-9a73-4306-9baa-ccfacdd68bee/100-2497 

Build Log : https://buildkite.com/chef-oss/chef-omnibus-software-main-verify/builds/1675

Reply:
@poorndm It is not downloading the actual software but source.html. Please find the below screenshot for the same. 
![Screenshot 2022-03-09 at 1 13 08 PM](https://user-images.githubusercontent.com/76031665/157395455-e0dafb9d-3cca-4e4f-b073-a8262563a835.png)
.

Reply:
Sure I will Take a look.